tests/ui/rfcs/rfc-2632-const-trait-impl/effects/minicore.rs

```diff
@@ -336,7 +336,7 @@ fn from_str(s: &str) -> Result<bool, ()> {
 }
 
 #[lang = "eq"]
-#[const_trait]
+// FIXME #[const_trait]
```

Comment:
What breaks here, out of curiosity? `from_str` should be `host = true`, so why can't we evaluate the `PartialEq` obligation in the match above?

Reply:
I think the only explanation here for this breaking is that we begun checking `==` method calls. Having this led to ICEs about mismatched number of generic arguments for `PartialEq`.